File: Python Implementation/9_1.py
class Solution(object):

    def prefixCount(self, words, pref):
        """
        :type words: List[str]
        :type pref: str
        :rtype: int
        """
        total = 0
        for i in words:
            if i.startswith(pref) :
                total += 1
        return total
    
sol = Solution()

l1 = ["hello","help","hi","heat","owl"]
print(sol.prefixCount(l1, "he"))

File: 9_1.cpp
/**
 * @file 9_1.cpp
 * @author Sricharan
 * @brief Given a list of words and a prefix, find the count of words that begin with that prefix
 * @version 1
 * @date 2025-01-09
 * 
 */

#include <iostream>
#include <vector>
#include <string_view>
using namespace std;

class Solution {
public:
    int prefixCount(vector<string>& words, string pref) {
        int total = 0;
        string_view pre(pref);
        for (int i = 0; i < words.size(); i++){
            if (words[i].length() >= pref.length()){
                if (pre == string_view(words[i].data(), pref.length())){
                    total++;
                }
            }
        }   
        return total;
    }   
};

int main(){
    Solution sol;
    vector<string> wordlist = {"hello", "hi", "heat","owl","help"};

    cout << sol.prefixCount(wordlist, "he");
}
